### ISSUE TITLE
auro-dialog: a11y - closed dialog not hidden from screen readers or tab order

### ISSUE THREAD

**Issue by geoffrich** [AlaskaAirlines/auro-dialog#6]
## Describe the bug
Despite not appearing on screen, the closed dialog is still read out to screen readers when navigating the page. It also appears that elements in the dialog are receiving focus when tabbing.

## To Reproduce
1. Go to the dialog example on the Auro doc site with a screen reader
2. Navigate to the dialog open button.
3. Continue navigating down the page. You will hear the contents of the dialog read out

Here is [a .zip file containing a video of the experience](https://github.com/AlaskaAirlines/auro-dialog/files/5283979/voiceover-playback.zip) on iOS with Voiceover.

## Expected behavior
The contents of a closed dialog are not read out to screen readers or focusable when tabbing.

## Smartphone (please complete the following information):
 - Device: iPhone 8
 - OS: iOS 13.6
 - Browser: Firefox (i.e. Safari) with Voiceover

## Additional context
Interestingly, NVDA/Firefox only read out the dialog header, not the rest of the dialog.

**Comment by geoffrich** [comment]
This could be solved by toggling `aria-hidden` and `inert` on the modal. We'd need to [polyfill inert](https://github.com/WICG/inert) and test to make sure it works as expected.

**Comment by geoffrich** [comment]
@blackfalcon I see that the PR you linked adds aria-hidden but I don't think it does anything to prevent tabbing into the modal when closed. Have you tested if you can still tab to the hidden modal?

**Comment by blackfalcon** [comment]
@geoffrich the aria-hidden does not prevent tabbing. The additional function is there for applying display none after the animation actually does that.
